src/containers/Tenant/Schema/SchemaViewer/SchemaViewer.tsx

```diff
@@ -19,15 +27,74 @@ const SchemaViewerColumns = {
     key: 'Key',
     type: 'Type',
     notNull: 'NotNull',
+    familyName: 'Family',
+    preferredPoolKind: 'Media',
+    columnCodec: 'Compression',
 };
 
 interface SchemaViewerProps {
-    keyColumnIds?: number[];
-    columns?: TColumnDescription[];
+    className?: string;
     type?: EPathType;
+    path?: string;
+    withFamilies?: boolean;
+}
+
+function prepareOlapTableSchema(tableSchema: TColumnTableDescription = {}) {
+    const {Name, Schema} = tableSchema;
+
+    if (Schema) {
+        const {Columns, KeyColumnNames} = Schema;
+        const KeyColumnIds = KeyColumnNames?.map((name: string) => {
+            const column = Columns?.find((el) => el.Name === name);
+            return column?.Id;
+        }).filter((id): id is number => id !== undefined);
+
+        return {
+            Columns,
+            KeyColumnNames,
+            Name,
+            KeyColumnIds,
+        };
+    }
+
+    return {
+        Name,
+    };
 }
 
-export const SchemaViewer = ({keyColumnIds = [], columns = [], type}: SchemaViewerProps) => {
+function formatColumnCodec(codec?: EColumnCodec) {
+    if (!codec) return null;
+    if (codec === EColumnCodec.ColumnCodecPlain) return 'None';
+    return codec.replace('ColumnCodec', '').toLocaleLowerCase();
+}
+
+export const SchemaViewer = ({className, type, path, withFamilies}: SchemaViewerProps) => {
+    const {data, loading} = useTypedSelector((state) => state.schema);
+    const currentObjectData = path ? data[path] : undefined;
+
+    let keyColumnIds: number[] = [];
+    let columns: TColumnDescription[] = [];
+
+    if (isTableType(type) && isColumnEntityType(type)) {
+        const description = currentObjectData?.PathDescription?.ColumnTableDescription;
+        const columnTableSchema = prepareOlapTableSchema(description);
+        keyColumnIds = columnTableSchema.KeyColumnIds ?? [];
+        columns = columnTableSchema.Columns ?? [];
+    } else if (isExternalTable(type)) {
+        columns = currentObjectData?.PathDescription?.ExternalTableDescription?.Columns ?? [];
+    } else {
+        keyColumnIds = currentObjectData?.PathDescription?.Table?.KeyColumnIds ?? [];
+        columns = currentObjectData?.PathDescription?.Table?.Columns ?? [];
+    }
+
+    const families =
+        currentObjectData?.PathDescription?.Table?.PartitionConfig?.ColumnFamilies?.reduce<
+            Record<number, TFamilyDescription>
+        >((acc, family) => {
+            if (family.Id) acc[family.Id] = family;
+            return acc;
+        }, {}) ?? {};
+
     // Keys should be displayd by their order in keyColumnIds (Primary Key)
     const keyColumnsOrderValues = React.useMemo(() => {
```

Comment:
Do you really need `useMemo` this place?

Reply:
You're right. Removed.